File: C++/SPR.cpp
#include <fstream>
//#include <algorithm>

short* v, n;

bool sumExists(short sum){
  if(sum == 0) return true;

  for(short i=0; i<n; i++)
    if(v[i] <= sum && sumExists(sum - v[i])) return true;

  return false;
}

short iteration(){
  //std::sort(v, v+n);
  for(short found=99; found>0; found--)
    if(!sumExists(found)) return found;
  return 0;
}

int main(){
  std::ifstream in("input.txt");
  in>>n;

  v = new short[n];
  for(short i=0; i<n; i++) in>>v[i];
  in.close();

  std::ofstream out("output.txt");
  out<<iteration();
  out.close();
}

File: Python/SPR.py
v = []

def sumExists(sum):
    if sum == 0:
        return True

    for i in v:
        if i <= sum and sumExists(sum - i):
            return True

    return False

def iteration():
    v.sort()
    for i in range(99, 1, -1):
        if not sumExists(i):
            return i
    return 0

f = open("input.txt", 'r')

next(f)

for i in f:
    v.append(int(i))

f = open("output.txt", 'w')
f.write(str(iteration()))
